### PR TITLE
Add atamahane（頭跳） compatibility code

### PR DESCRIPTION
**Pull request by rasca0027**
Add atamahane and allow multiple ron players code for Kyoku.
Fix end kyoku and ryuukyoku logic.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have run the tests and make sure they pass
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tests/test_points.py [hunk 1]
```diff
@@ -17,48 +17,48 @@ def setUp(self):
...
-        self.kyoku.winner = self.player_2
+        self.kyoku.winners = [self.player_2]
         self.kyoku.honba = 1
         self.kyoku.apply_points(4, 25, True)
         self.assertEqual(self.player_1.points, 21_700)
         self.assertEqual(self.player_2.points, 31_700)
-        self.assertEqual(self.player_3.points, 23_300)
-        self.assertEqual(self.player_4.points, 23_300)
+        self.assertEqual(self.player_3.points, 23_400)
+        self.assertEqual(self.player_4.points, 23_400)
```
Comment (rasca0027) [inline tests/test_points.py:58]: 其實這個test case希望大家幫忙看一下 我沒腦力了

File: mahjong/helpers.py [hunk 2]
```diff
@@ -139,3 +139,18 @@ def is_pon(tile_set: List[Tile]) -> bool:
...
+
+def get_atamahane_winner(discard_pos: int, winners: List[int]):
+    """This function determines the actual winner according to order of
+    seating closest to discarder.
+    Return:
+        winner_pos: List[int]
+    """
+    if not discard_pos:
+        return winners[0]
+    seats = [x + 4 for x in winners]
```
Comment (rasca0027) [inline mahjong/helpers.py:152]: [x + 4 if x < discard_pos else x for x in winners] 

`discard_pos` -> ？

File: mahjong/kyoku.py [hunk 3]
```diff
@@ -351,37 +362,36 @@ def start(self):
 
         # 莊家 oya draw flow
         turn = Turn(self.players, self.tile_stack, self.logger)
-        state, discard_tile, discard_pos = turn.draw_flow(self.oya_player)
+        state, discard_tile, discard_pos, act = turn.draw_flow(self.oya_player)
         # Tenhoo
         while state == 0:
             state, discard_tile, discard_pos = turn.discard_flow(
```
Comment (rasca0027) [inline mahjong/kyoku.py:368]: 少一個act

File: mahjong/kyoku.py [hunk 4]
```diff
@@ -351,37 +362,36 @@ def start(self):
...
             else:
                 return False, self.kyotaku, 0
 
         else:
-            # TODO: 三家和流局: if len(sef.winner) == 3
-            # TODO: Check Yaku and calculate the amount.
-            tsumo = False  # placeholder, 不然玉米片一直吐錯誤
-            loser = None
+            tsumo = act == Action.TSUMO
+            loser = self.players[discard_pos]
```
Comment (rasca0027) [inline mahjong/kyoku.py:380]: discard可能None

File: mahjong/kyoku.py [hunk 5]
```diff
@@ -351,37 +362,36 @@ def start(self):
...
+            self.winners = [self.players[pos] for pos in turn.winners]
```
Comment (rasca0027) [inline mahjong/kyoku.py:381]: Turn裡面的winners改成叫winner_pos之類的

File: mahjong/kyoku.py [hunk 6]
```diff
@@ -450,31 +460,44 @@ def apply_points(self,
...
+                winner.points += tmp_pt
+        else:  # ron
+            for winner in self.winners:
+                if winner == self.oya_player:
+                    winner.points += roundup(pt * 6)
+                    loser.points -= roundup(pt * 6)
+                else:  # 子家
+                    winner.points += roundup(pt * 4)
+                    loser.points -= roundup(pt * 4)
+            if not ryuukyoku:
```
Comment (rasca0027) [inline mahjong/kyoku.py:497]: 流局滿貫好像是算在自摸？？

Comment (rasca0027) [reply mahjong/kyoku.py:497]: 考慮兩個人流局滿貫

File: tests/test_points.py (hunk 1 above)
Comment (rasca0027) [reply tests/test_points.py:58]: 晚點再看看ＱＱ

File: tests/test_turn.py [hunk 7]
```diff
@@ -44,7 +47,8 @@ def test_ankan(self):
             ((Action.NOACT, Naki.NONE), Tile(0, 1))]
         self.turn.kan_flow = MagicMock(return_value=0)
 
-        state, discard_tile, discard_pos = self.turn.draw_flow(self.player_1)
+        state, discard_tile, discard_pos, act = self.turn.draw_flow(
+            self.player_1)
```
Comment (rasca0027) [inline tests/test_turn.py:51]: testcase記得比較第四個action

File: mahjong/helpers.py (hunk 2 above)
Comment (ktc312) [reply mahjong/helpers.py:152]: #119

File: tests/test_points.py (hunk 1 above)
Comment (ktc312) [reply tests/test_points.py:58]: #118

**Review by HappyBall — APPROVED** [review]

**Pull request merged**